vue/components/ui/molecules/bezier-curve/bezier-curve.vue

```diff
@@ -0,0 +1,377 @@
+}
+
+.bezier-curve .point:hover {
+    cursor: pointer;
+    filter: drop-shadow(0px 0px 3px #007bff);
+}
+
+.bezier-curve .point:active {
+    cursor: move;
+}
+</style>
+
+<script>
+export const BezierCurve = {
+    name: "bezier-curve",
+    props: {
+        /**
+         * Bezier curve endpoint 1 X coordinate.
+         */
+        x1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 1 Y coordinate.
+         */
+        y1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 2 X coordinate.
+         */
+        x2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 2 Y coordinate.
+         */
+        y2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 X coordinate.
+         */
+        cx1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 Y coordinate.
+         */
+        cy1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 X coordinate.
+         */
+        cx2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 Y coordinate.
+         */
+        cy2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * The number of pixels of the svg viewport
+         * in the format "`width`x`height`".
+         */
+        resolution: {
+            type: String,
+            default: "1024x1024"
+        },
+        /**
+         * Visibilty of Bezier curve control points.
+         */
+        showPoints: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * The interval at which the new coordinates
+         * will be emited when dragging.
+         */
+        emitInterval: {
+            type: Number,
+            default: 250
+        },
+        /**
+         * The decimal places to define the numerical precision
+         * when emitting the position of control and end points.
+         */
+        roundPlaces: {
+            type: Number,
+            default: 2
+        },
+        /**
+         * Weather or not the control points are disabled
+         * and unable to control.
+         */
+        disabled: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data: function() {
+        return {
+            bezier: {
+                x1: this.x1,
+                y1: this.y1,
+                x2: this.x2,
+                y2: this.y2,
+                cx1: this.cx1,
+                cx2: this.cx2,
+                cy1: this.cy1,
+                cy2: this.cy2
+            },
+            dragging: null,
+            reactiveEmit: null

```

Comment:
call this `emitTimeout` (we usually do it this way for timeouts, to make it clear that this holds a function)

Reply:
emitInterval, I mean (you can change the existing one to emitIntervalMs or smth like that to avoid collision)